### ISSUE TITLE
URI Resource Refactor

### ISSUE THREAD

**Issue by roidriscoll** [makersacademy/pipekit#22]
Currently we are doing a bad job of pluralizing URI resources, which has led to an annoying bug while trying to implement the Activities feature in Pipekit (#21). At the moment I have two options - one to do something clever to work out the correct resource name (or include a gem that does it for me) - or add an attribute to a class which is it's correctly-pluralized name. I am in favour of this last approach as it seems the safest option.

**Comment by roidriscoll** [comment]
Had to pause work on this Friday to deal with some of the issues Marketing were experiencing with broken links.

**Comment by roidriscoll** [comment]
This took all of Monday, but settled on the approach of adding a constant to each class that represents a resource on PipeDrive. Will try and get one of the coaches to review the PR.

**Comment by roidriscoll** [comment]
PR approved, and all seemed well - until I tried to implement this with Applications and realised that the simple approach is not feasible, so had to revisit this. I have tried to keep the constant-in-class approach and add a "Resource Label" object which allows the app to retrieve either a singular or pluralized resource name. I suspect this will need to be refactored again once I have it working, but going for mvp first. Also upgrading the ticket from a 2, to a 3.
